src/Microsoft.AspNetCore.Mvc.Modules.Hosting/Mvc/Razor/ModuleViewLocationExpander.cs

```diff
@@ -16,8 +23,23 @@ public void PopulateValues(ViewLocationExpanderContext context)
         {
             var result = new List<string>();
 
-            result.Add("/Modules/{2}/Views/{1}/{0}.cshtml");
-            result.Add("/Modules/{2}/Views/Shared/{0}.cshtml");
+            var extensionManager = context.ActionContext.HttpContext.RequestServices.GetService<IExtensionManager>();
+            var shellFeaturesManager = context.ActionContext.HttpContext.RequestServices.GetService<IShellFeaturesManager>();
+
+            var availableFeatures = extensionManager.GetExtensions().Features;
+            var enabledFeatures = shellFeaturesManager.GetEnabledFeaturesAsync().Result;
+
+            var features = availableFeatures.Where(af => enabledFeatures.Any(ef => af.Id == ef.Id));
+            var extensions = features.Where(f => f.Id == f.Extension.Id).Select(f => f.Extension).Reverse();
```

Comment:
We grab extensions from main features (not from sub features) to keep the order based on these main features which provide through their extension physical view subpaths.

Note: just commited my last updates.